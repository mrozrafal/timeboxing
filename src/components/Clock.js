import React from "react";

function Clock ({className,  minutes, seconds }) {      
    return  (
<h2 className={"clock " + className }>
      Pozosta≈Ço
      <div>
      <div className="clock__minutes">{formater(minutes)}</div>
      <div className="clock__separator">:</div>
      <div className="clock__seconds">{formater(seconds)}</div>
      </div>
      
    </h2>
    )
    
  }
  function formater (props){
    let formater;
      if (props < 0){
      formater = `00`;
      }else if(props < 10){
        formater = `0${props.toString()}`;
      }else{
         formater = props;
      }            
     return `${formater}`;
    }

  export default Clock;