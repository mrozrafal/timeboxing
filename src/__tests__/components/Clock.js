import React from "react";
import Clock from "../../components/Clock";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";


var root = null;
var clockRenderer = null;
describe('<Clock />', () => {
    describe('when given minutes and seconds (DOM)', () => {
        beforeEach(() => {
            root = document.createElement("div");
            ReactDOM.render(
                <Clock minutes={25} seconds={50} />, root
            )
        });


        it('renders propertly', () => {
            expect(root.childNodes[0].nodeName).toEqual("H2")
            expect(root.childNodes[0].className).toMatch(/Clock/)
            expect(root.childNodes[0].textContent).toMatch(/25:50/)
        });
        it('renders an h2 element', () => {
            expect(root.childNodes[0].nodeName).toEqual("H2")
        });
        it('sets a Clock className', () => {
            expect(root.childNodes[0].className).toMatch(/Clock/)
        });
        it('renders time propertly', () => {
            expect(root.childNodes[0].textContent).toMatch(/25:50/)
        });
    });

    describe('when given minutes and seconds (TestRenderer)', () => {
        beforeEach(() => {
            clockRenderer = renderer.create(
                <Clock minutes={25} seconds={50} />
            )
        });


        it('renders propertly', () => {
            console.log(clockRenderer.toJSON())
            expect(clockRenderer.toJSON()).toMatchSnapshot()
        });
        it('renders an h2 element', () => {
            expect(clockRenderer.toJSON().type).toEqual("h2")
        });
        it('sets a Clock className', () => {
            expect(clockRenderer.toJSON().props).toMatchObject({"className": expect.stringMatching(/Clock/)})
        });
        it('renders time propertly', () => {
            expect(clockRenderer.toJSON().children).toEqual(expect.arrayContaining(["25", "50"]))
        });
    });

    it('sets className to empty string if not given anything else', () => {
        expect(<Clock minutes={25} seconds={50} />).toEqual(<Clock className="" minutes={25} seconds={50} />)
    });
});