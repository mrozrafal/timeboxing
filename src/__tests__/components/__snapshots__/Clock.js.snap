// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> when given minutes and seconds (TestRenderer) renders propertly 1`] = `
<h2
  className="Clock "
>
  Pozosta≈Ço 
  25
  :
  50
</h2>
`;
